src/smolagents/agents.py

```diff
@@ -88,11 +88,12 @@ class ActionStep(AgentStepLog):
     tool_calls: List[ToolCall] | None = None
     start_time: float | None = None
     end_time: float | None = None
-    step: int | None = None
+    step_number: int | None = None
```

Comment:
I noticed this is a breaking change. Do we want to keep compatibility with older versions?